### PR TITLE
adding supporting data properties, fixes #1074

### LINKED ISSUE #1074: supporting study results for COHD
**Is your feature request related to a problem? Please describe.**
Matt B. helped us to develop a 2-level `supporting study result` structure to model COHD data, but most of the proposed elements are not defined in Biolink yet. 
[Google doc](https://docs.google.com/document/d/1YHsSSn8ML0iPJg9-3XsqMq-1eiFiPih8qij7h-26BKo/edit#heading=h.j4lrz7khxrbl) describing the model. Matt's [examples ](https://github.com/NCATSTranslator/Evidence-Provenance-Confidence-Working-Group/blob/master/attribute_epc_examples/COHD_TRAPI1.2_nested_attributes.yml)(see the example set at the bottom)

**What working group (or team) did this request originate from?**
Clinical Data Services (COHD)

**Note:** This is relevant for members of NCATS Translator.

**Describe the solution you'd like**
The following may need to be defined in Biolink:
* biolink:supporting_study_result
* biolink:ConceptCountAnalysisResult
* biolink:concept_pair_count
* biolink:concept_count_subject
* biolink:concept_count_object
* biolink:supporting_dataset
* biolink:ChiSquaredAnalysisResult
* biolink:unadjusted_p-value
* biolink:bonferonni_adjusted_p-value
* biolink:Observed-ExpectedFrequencyAnalysisResult
* biolink:expected_count
* biolink:ln_ratio
* biolink:ln_ratio_99_confidence_interval
* biolink:RelativeFrequencyAnalysisResult
* biolink:relative_frequency_subject
* biolink:relative_freq_subject_confidence_interval
* biolink:relative_frequency_object
* biolink:relative_freq_object_confidence_interval

**Additional information to support this request** (optional)
We've recently implemented this model in the COHD's dev endpoint at https://cohd.io/api/query and we'd like to push these changes out to the ITRB environments soon (ideally before the Aug 29 code freeze). From our perspective, we're not in a rush to have these defined immediately in Biolink as it appears that reasoners are still able to utilize the data (e.g., ARAX UI is able to display the results). We are open to guidance on how to move forward with this. 

Sample query that can be sent to COHD to see results:
```
{
    "message": {
        "query_graph": {
            "nodes": {
                "subj": {
                    "ids": ["MONDO:0009061"]
                },
                "obj": {
                    "categories": ["biolink:DiseaseOrPhenotypicFeature"]
                }
            },
            "edges": {
                "e0": {
                    "subject": "subj",
                    "object": "obj",
                    "predicates": ["biolink:has_real_world_evidence_of_association_with"]
                }
            }
        }
    }
}
```

Sample attributes from a single edge:
```
                    "attributes": [
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:original_knowledge_source",
                            "description": "The COHD KP defines associations between biomedical concepts based on statistical analysis of clinical/EHR data.",
                            "value": "infores:cohd",
                            "value_type_id": "biolink:InformationResource",
                            "value_url": "http://cohd.io/api/query"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_dataset",
                            "description": "Dataset ID within COHD",
                            "original_attribute_name": "dataset_id",
                            "value": "COHD:dataset_1",
                            "value_type_id": "EDAM:data_1048"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_study_result",
                            "attributes": [
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:concept_pair_count",
                                    "description": "Observed concept count between the pair of subject and object nodes",
                                    "original_attribute_name": "concept_pair_count",
                                    "value": 449,
                                    "value_type_id": "EDAM:data_0006"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:concept_count_subject",
                                    "description": "Observed concept count of the subject node",
                                    "original_attribute_name": "concept_count_subject",
                                    "value": 968,
                                    "value_type_id": "EDAM:data_0006"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:concept_count_object",
                                    "description": "Observed concept count of the object node",
                                    "original_attribute_name": "concept_count_object",
                                    "value": 556,
                                    "value_type_id": "EDAM:data_0006"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:supporting_dataset",
                                    "description": "Dataset ID within COHD",
                                    "original_attribute_name": "dataset_id",
                                    "value": "COHD:dataset_1",
                                    "value_type_id": "EDAM:data_1048"
                                }
                            ],
                            "description": "A study result describing the initial count of concepts",
                            "value": null,
                            "value_type_id": "biolink:ConceptCountAnalysisResult"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_study_result",
                            "attributes": [
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:unadjusted_p-value",
                                    "description": "Chi-square p-value, unadjusted. http://cohd.io/about.html",
                                    "original_attribute_name": "p-value",
                                    "value": 1e-12,
                                    "value_type_id": "EDAM:data_1669",
                                    "value_url": "http://edamontology.org/data_1669"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:bonferonni_adjusted_p-value",
                                    "description": "Chi-square p-value, Bonferonni adjusted by number of pairs of concepts. http://cohd.io/about.html",
                                    "original_attribute_name": "p-value adjusted",
                                    "value": 1e-12,
                                    "value_type_id": "EDAM:data_1669",
                                    "value_url": "http://edamontology.org/data_1669"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:supporting_dataset",
                                    "description": "Dataset ID within COHD",
                                    "original_attribute_name": "dataset_id",
                                    "value": "COHD:dataset_1",
                                    "value_type_id": "EDAM:data_1048"
                                }
                            ],
                            "description": "A study result describing a chi-squared analysis on a single pair of concepts",
                            "value": null,
                            "value_type_id": "biolink:ChiSquaredAnalysisResult"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_study_result",
                            "attributes": [
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:expected_count",
                                    "description": "Calculated expected count of concept pair. For ln_ratio. http://cohd.io/about.html",
                                    "original_attribute_name": "expected_count",
                                    "value": 0.3006024806317585,
                                    "value_type_id": "EDAM:operation_3438"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:ln_ratio",
                                    "description": "Observed-expected frequency ratio. http://cohd.io/about.html",
                                    "original_attribute_name": "ln_ratio",
                                    "value": 7.308989437171575,
                                    "value_type_id": "EDAM:data_1772"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:ln_ratio_99_confidence_interval",
                                    "description": "Observed-expected frequency ratio 0.99% confidence interval",
                                    "original_attribute_name": "ln_ratio_confidence_interval",
                                    "value": [
                                        7.1447659245560216,
                                        7.455795361004793
                                    ],
                                    "value_type_id": "EDAM:data_0951"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:supporting_dataset",
                                    "description": "Dataset ID within COHD",
                                    "original_attribute_name": "dataset_id",
                                    "value": "COHD:dataset_1",
                                    "value_type_id": "EDAM:data_1048"
                                }
                            ],
                            "description": "A study result describing an observed-expected frequency anaylsis on a single pair of concepts",
                            "value": null,
                            "value_type_id": "biolink:Observed-ExpectedFrequencyAnalysisResult"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_study_result",
                            "attributes": [
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:relative_frequency_subject",
                                    "description": "Relative frequency, relative to the subject node. http://cohd.io/about.html",
                                    "original_attribute_name": "relative_frequency_subject",
                                    "value": 0.46384297520661155,
                                    "value_type_id": "EDAM:data_1772"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:relative_freq_subject_confidence_interval",
                                    "description": "Relative frequency (subject) 0.99% confidence interval",
                                    "original_attribute_name": "relative_freq_subject_confidence_interval",
                                    "value": [
                                        0.3765490943755958,
                                        0.5680539932508436
                                    ],
                                    "value_type_id": "EDAM:data_0951"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:relative_frequency_object",
                                    "description": "Relative frequency, relative to the object node. http://cohd.io/about.html",
                                    "original_attribute_name": "relative_frequency_object",
                                    "value": 0.8075539568345323,
                                    "value_type_id": "EDAM:data_1772"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:relative_freq_object_confidence_interval",
                                    "description": "Relative frequency (object) 0.99% confidence interval",
                                    "original_attribute_name": "relative_freq_object_confidence_interval",
                                    "value": [
                                        0.63915857605178,
                                        1.0181451612903225
                                    ],
                                    "value_type_id": "EDAM:data_0951"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:supporting_dataset",
                                    "description": "Dataset ID within COHD",
                                    "original_attribute_name": "dataset_id",
                                    "value": "COHD:dataset_1",
                                    "value_type_id": "EDAM:data_1048"
                                }
                            ],
                            "description": "A study result describing a relative frequency anaylsis on a single pair of concepts",
                            "value": null,
                            "value_type_id": "biolink:RelativeFrequencyAnalysisResult"
                        }
                    ]
```

Also, Text Mining Provider uses a similar structure and may need similar support. See the Google doc linked above, and [Bill's slides from the KP of the Month](https://docs.google.com/presentation/d/1e5ftEHO4x_PkW4Q_mHBmwVqEzaXKLyMYJFs52nw4UoE/edit#slide=id.g147d2142526_0_302) presentation. 

**Tag relevant members for discussion**
@mbrush @bill-baumgartner

### PR DESCRIPTION
**Pull request by sierra-moxon**

### CODE REVIEW

File: biolink-model.yaml [hunk 1]
```diff
@@ -6512,6 +6516,7 @@ slots:
     multivalued: true
 
   supporting data source:
+    aliases: ['supporting dataset']
```
Comment (mbrush) [inline biolink-model.yaml:6519]: See my comments here: https://github.com/biolink/biolink-model/issues/1074#issuecomment-1228937425

File: biolink-model.yaml [hunk 2]
```diff
@@ -6549,6 +6555,170 @@ slots:
...
+  expected count:
+    is_a: association slot
+    description: >-
+      The expected (calculated) number of instances in a dataset/cohort whose records contain both the subject and 
+      object concept of an association if the subject and object concepts are independent.
+
+  relative frequency subject:
+    is_a: association slot
+    description: >-
+      The frequency at which subject and object concepts co-occur in EHR
```
Comment (mbrush) [inline biolink-model.yaml:6592]: I would remove 'EHR' from this description - to make this property more general / re-usable.

File: biolink-model.yaml [hunk 3]
```diff
@@ -6549,6 +6555,170 @@ slots:
...
+      records within a dataset/cohort, relative to the frequency at which the subject
+      concept appears in these same records.
+    range: float
+    examples:
+      - value: 0.01840490798
+
+  relative frequency object:
+    is_a: association slot
+    description: >-
+      The frequency at which subject and object concepts co-occur in EHR
```
Comment (mbrush) [inline biolink-model.yaml:6602]: As above - remove 'EHR' to make this property more reusable.

File: biolink-model.yaml [hunk 4]
```diff
@@ -6549,6 +6555,170 @@ slots:
...
+
+  relative frequency object confidence interval:
+    is_a: association slot
+    description: >-
+      The 99% confidence interval for the relative_frequency_object calculation
+      (i.e. the range of values within which the true value has a 99% chance of falling)
+    range: string
+
+  adjusted p value:
+    is_a: association slot
```
Comment (mbrush) [inline biolink-model.yaml:6622]: Seems like we should have a hierarchy of more specific p-value flavors.  a generic root 'p-value', with child slots for 'adjusted p-value', 'unadjusted p-value', 'bonferoni adjusted p-value' (which may actually go under 'adjusted p-value') . . . thoughts?  Maybe keep flat for now, and save this for next time?

File: biolink-model.yaml [hunk 5]
```diff
@@ -6549,6 +6555,170 @@ slots:
...
+      statistical tests are being performed simultaneously on a single data set. To perform a Bonferroni 
+      correction, divide the critical P value (α) by the number of comparisons being made.  P is always italicized and 
+      capitalized. The actual P value* should be expressed (P=. 04) rather than expressing a statement of inequality 
+      (P<. 05), unless P<.
+    range: float
+    examples:
+      - value: 0.018
+
+  supporting text:
+    is_a: association slot
```
Comment (mbrush) [inline biolink-model.yaml:6644]: It occurred to me that many of the properties we are adding, like this one, will actually hang from a StudyResult object, not an Association - which I think means that technically they are not 'association slots', right?  I think that StudyResult object will always hang form Associations, so these slots are association_slots  in a indirect sense . . . . which may be enough justification to place them under 'association slot' int he biolink slot hierarchy.

File: biolink-model.yaml [hunk 6]
```diff
@@ -6549,6 +6555,170 @@ slots:
...
+    description: The segment of text from a document that supports the mined assertion.
+    range: string
+    examples:
+      - value: The administration of 50g/ml bupivacaine promoted maximum breast cancer.
+
+  supporting document:
+    is_a: association slot
+    description: The document that contains the supporting text of a Text Mining Result.
```
Comment (mbrush) [inline biolink-model.yaml:6652]: I would generalize the definition here as well - as we will need a slot like this in other contexts.  In our draft EPC model we have proposed a 'has supporting document' property with the following definition: "A publication or other document that reports the statement expressed in an Association, or provides information used as evidence supporting this statement.".  

If Bill / Mike agree this definition applies here (IMO it does), let's use something like this.

File: biolink-model.yaml [hunk 7]
```diff
@@ -6998,6 +7168,38 @@ classes:
     is_a: named thing
     abstract: true
 
+  supporting study result set:
+    abstract: true
+    is_a: information content entity
+    description: >-
+      A result of a supporting study.
```
Comment (mbrush) [inline biolink-model.yaml:7175]: not sure what a 'supporting study result set' is . . . the definition here doesn't help.

Comment (sierra-moxon) [reply biolink-model.yaml:7190]: done!

File: biolink-model.yaml (hunk 6 above)
Comment (sierra-moxon) [reply biolink-model.yaml]: done

File: biolink-model.yaml (hunk 5 above)
Comment (sierra-moxon) [reply biolink-model.yaml:6657]: yep, we could have another high-level grouping slot "study result slot" as a sibling to association_slot perhaps; in particular if we need it for technical reasons (grouping all these together).  Holding off for now.

File: biolink-model.yaml (hunk 1 above)
Comment (sierra-moxon) [reply biolink-model.yaml]: removed alias and added a second slot for 'study data set'

File: biolink-model.yaml (hunk 3 above)
Comment (sierra-moxon) [reply biolink-model.yaml]: fixed.

File: biolink-model.yaml (hunk 2 above)
Comment (sierra-moxon) [reply biolink-model.yaml]: done

File: biolink-model.yaml (hunk 5 above)
Comment (mbrush) [reply biolink-model.yaml:6657]: Yeah, I would hold off.  I think many of these properties could hang directly from an Association in some cases, or be used within a Study Result in other cases.  Fine for no I think to not worry about this nuance, and just call them all Association slots (even if within an Association structure they are nested in some object that hangs from the Association directly)

**Pull request merged**